x/ibc-transfer/types/coin.go

```diff
@@ -0,0 +1,51 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// SenderChainIsSource returns false if the denomination originally came
+// from the receiving chain and true otherwise.
+func SenderChainIsSource(sourcePort, sourceChannel, denom string) bool {
+	// This is the prefix that would have been prefixed to the denomination
+	// on sender chain IF and only if the token originally came from the
+	// receiving chain.
+
+	voucherPrefix := GetDenomPrefix(sourcePort, sourceChannel)
+	return !strings.HasPrefix(denom, voucherPrefix)
+}
+
+// ReceiverChainIsSource returns true if the denomination orgininally came
+// from the receiving chain and false otherwise.
+func ReceiverChainIsSource(prefix, denom string) bool {
```

Comment:
It's weird to me that this function is symmetric with `SenderChainIsSource` but has a different function signature. I think that can be confusing to developers.

I think this should have same signature
```suggestion
func ReceiverChainIsSource(sourcePort, sourceChannel, denom string) bool {
```

Creating the prefix twice is not a big deal, and worth the improvement in dev UX imo

Reply:
I'm just going to call `SenderChainIsSource` and return with a `!`